graphstorm-processing/graphstorm_processing/data_transformations/dist_transformations/dist_numerical_transformation.py

```diff
@@ -0,0 +1,349 @@
+        Valid values are "mean", "median" and "most_frequent".
+    """
+
+    def __init__(self, cols: Sequence[str], normalizer: str, imputer: str) -> None:
+        super().__init__(cols)
+        self.cols = cols
+        self.shared_norm = normalizer
+        # Spark uses 'mode' for the most frequent element
+        self.shared_imputation = "mode" if imputer == "most_frequent" else imputer
+
+    def apply(self, input_df: DataFrame) -> DataFrame:
+        logging.debug(
+            "Applying normalizer: %s, imputation: %s", self.shared_norm, self.shared_imputation
+        )
+
+        imputed_df = apply_imputation(self.cols, self.shared_imputation, input_df)
+        scaled_df = apply_norm(self.cols, self.shared_norm, imputed_df)
+
+        # TODO: Figure out why the transformation is producing Double values, and switch to float
+        return scaled_df
+
+    @staticmethod
+    def get_transformation_name() -> str:
+        return "DistNumericalTransformation"
+
+
+class DistMultiNumericalTransformation(DistNumericalTransformation):
+    """Transformation to apply missing value imputation and various forms of normalization
+     to a multi-column numerical input, where the input is a string separated by a delimiter.
+
+    Parameters
+    ----------
+    cols : Sequence[str]
+        The list of columns to apply the transformations on.
+    separator: str
+        The separator string that divides the string values.
+    normalizer : str
+        The normalization to apply to the columns.
+        Valid values are "none", "min-max", and "standard".
+    imputer : str
+        The type of missing value imputation to apply to the column.
+        Valid values are "mean", "median" and "most_frequent".
+    """
+
+    def __init__(
+        self, cols: Sequence[str], separator: Optional[str], normalizer: str, imputer: str
+    ) -> None:
+        assert (
+            len(cols) == 1
+        ), "DistMultiNumericalTransformation only supports one column at a time."
+        super().__init__(cols, normalizer, imputer)
+        self.multi_column = cols[0]
+
+        self.separator = separator
+        # Keep the original separator to split in pure Python
+        self.original_separator = self.separator
+        # Spark's split function uses a regexp so we need to escape
+        # special chars to be used as separators
+        if self.separator in SPECIAL_CHARACTERS:
+            self.separator = f"\\{self.separator}"
+
+    @staticmethod
+    def get_transformation_name() -> str:
+        return "DistMultiNumericalTransformation"
+
+    @staticmethod
+    def apply_norm_vector(vector_col: str, shared_norm: str, vector_df: DataFrame) -> DataFrame:
+        """
+        Applies normalizer column-wise with a single vector column as input.
+        """
+        other_cols = list(set(vector_df.columns).difference([vector_col]))
+
+        if shared_norm == "none":
+            scaled_df = vector_df
+        elif shared_norm == "min-max":
```

Comment:
Why not using `apply_norm(cols: Sequence[str], shared_norm: str, imputed_df: DataFrame) -> DataFrame`?

Reply:
I'll double-check I think it had to do with retaining all columns.

Reply:
Basically the implementation can be shared.

Reply:
Verified, this, we want to avoid using `apply_norm` here because it tries to use VectorAssembler to concat the existing columns, but our input is already in Vector form here so we're skipping one pass through the data this way.